lib/srv/alpnproxy/auth/auth_proxy.go

```diff
@@ -113,13 +114,21 @@ func (s *AuthProxyDialerService) dialLocalAuthServer(ctx context.Context) (net.C
 	if len(authServers) == 0 {
 		return nil, trace.NotFound("empty auth servers list")
 	}
-	//TODO(smallinksy) Better support for HA. Add dial retry on auth network errors.
-	authServerIndex := rand.Intn(len(authServers))
-	conn, err := net.Dial("tcp", authServers[authServerIndex].GetAddr())
-	if err != nil {
-		return nil, trace.Wrap(err)
+	var errors []string
```

Comment:
You could make this `var errors []error`.

Then use `errors = append(errors, fmt.Errorf("%s: %w", addr, err)`.

And then return `trace.NewAggregate(errors)`

Reply:
thanks! I haven't spotted `trace.NewAggregate`